### PR TITLE
replaces HashSet with List for WelderRefinable

### PR DESCRIPTION
**Pull request by BlueHNT**
# Description

This PR aims to turn a HashSet string? into List EntitySpawnEntry? for better and more detailed entry for junk system.

---

# TODO

- [x] Update RefiningSystem.cs to spawn items properly via EntitiySpawnEntry
- [x] Replace all instances of `WelderRefinableComponent` to match it's new format

---

### CODE REVIEW

File: Content.Server/Construction/Components/WelderRefinableComponent.cs [hunk 1]
```diff
@@ -11,12 +12,12 @@ namespace Content.Server.Construction.Components
     public sealed partial class WelderRefinableComponent : Component
     {
         [DataField("refineResult")]
```
Comment (Pspritechologist) [inline Content.Server/Construction/Components/WelderRefinableComponent.cs:14]:
```suggestion
        [DataField]
```

File: Content.Server/Construction/Components/WelderRefinableComponent.cs [hunk 2]
```diff
@@ -11,12 +12,12 @@ namespace Content.Server.Construction.Components
...
-        public HashSet<string>? RefineResult = new();
+        public List<EntitySpawnEntry>? RefineResult = new();
 
         [DataField("refineTime")]
         public float RefineTime = 2f;
 
-        [DataField("qualityNeeded", customTypeSerializer:typeof(PrototypeIdSerializer<ToolQualityPrototype>))]
+        [DataField("qualityNeeded", customTypeSerializer: typeof(PrototypeIdSerializer<ToolQualityPrototype>))]
         public string QualityNeeded = "Welding";
```
Comment (Pspritechologist) [inline Content.Server/Construction/Components/WelderRefinableComponent.cs:21]:
```suggestion
        [DataField]
        public ProtoId<ToolQualityPrototype> QualityNeeded = "Welding";
```

File: Content.Server/Construction/Components/WelderRefinableComponent.cs [hunk 3]
```diff
@@ -11,12 +12,12 @@ namespace Content.Server.Construction.Components
     public sealed partial class WelderRefinableComponent : Component
     {
         [DataField("refineResult")]
-        public HashSet<string>? RefineResult = new();
+        public List<EntitySpawnEntry>? RefineResult = new();
 
         [DataField("refineTime")]
```
Comment (Pspritechologist) [inline Content.Server/Construction/Components/WelderRefinableComponent.cs:17]:
```suggestion
        [DataField]
```

File: Content.Server/Construction/Components/WelderRefinableComponent.cs [hunk 4]
```diff
@@ -11,12 +12,12 @@ namespace Content.Server.Construction.Components
     public sealed partial class WelderRefinableComponent : Component
     {
         [DataField("refineResult")]
-        public HashSet<string>? RefineResult = new();
+        public List<EntitySpawnEntry>? RefineResult = new();
```
Comment (Pspritechologist) [inline Content.Server/Construction/Components/WelderRefinableComponent.cs:15]:
```suggestion
        public List<EntitySpawnEntry> RefineResult = new();
```

File: Content.Server/Construction/RefiningSystem.cs [hunk 5]
```diff
@@ -39,14 +40,9 @@ private void OnDoAfter(EntityUid uid, WelderRefinableComponent component, Welder
             EntityManager.DeleteEntity(uid);
 
             // spawn each result after refine
-            foreach (var result in component.RefineResult!)
+            foreach (var ent in EntitySpawnCollection.GetSpawns(component.RefineResult!))
```
Comment (Pspritechologist) [inline Content.Server/Construction/RefiningSystem.cs:43]:
```suggestion
            foreach (var ent in EntitySpawnCollection.GetSpawns(component.RefineResult))
```

**Review by Pspritechologist — CHANGES_REQUESTED** [review]
It seems like a lot of IDs got replaced with the same ID followed by `1`. Is this intentional?

File: Content.Server/Construction/Components/WelderRefinableComponent.cs (hunk 1 above)
Comment (Pspritechologist) [reply Content.Server/Construction/Components/WelderRefinableComponent.cs:14]: DataFields no longer need explicit strings to go along with them, so you should probably remove them whenever you modify a system. 

Then again that might just be EE brain, I dunno if this codebase has explicit rules on modifying upstream files or anything.

**Comment by BlueHNT** [comment]
> It seems like a lot of IDs got replaced with the same ID followed by `1`. Is this intentional?

Thank you for the help!

The IDs for the crystals was an oversight on my side.
As for the IDs for SheetMetal1 and such. It seems to pick up its entity states not the stacks. So it would pick up its whole stack instead of just the `amount:` number. Meaning a part where it would say
```C#
  - id: SheetMetal
    amount: 2
```
You would get 60 pieces instead of 2 pieces in a _stack
Another thing to perfect at some point I think :)

**Pull request merged**